internal/conduit/version.go

```diff
@@ -0,0 +1,107 @@
+package conduit
+
+import (
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+type Flags struct {
+	args *Args
+}
+
+type Args struct {
+	PipelineFile   string
+	ConnectorsPath string
+	DBPath         string
+	ProcessorsPath string
+}
+
+func NewFlags(fns ...func(*Args)) *Flags {
+	var args Args
+	for _, fn := range fns {
+		fn(&args)
+	}
+	return &Flags{args: &args}
+}
+
+func (f *Flags) ForVersion(ver string) []string {
+	constraints := map[string]string{
+		"v011": "< 0.12.0",
+		"v012": ">= 0.12.0, < 0.13.0",
+	}
+
+	sanitized, _ := strings.CutPrefix(ver, "v")
+	v, _ := semver.NewVersion(sanitized)
+
+	for key, rule := range constraints {
+		c, _ := semver.NewConstraint(rule)
+		if c.Check(v) {
+			switch key {
+			case "v011":
+				return f.v011()
+			case "v012":
+				return f.v012()
+			}
+		}
+	}
+	return f.v013()
```

Comment:
I think the default case is really an unexpected and this func should likely return and error to show that support is missing. Being explicit will help.